server/src/main/java/org/elasticsearch/node/NodeService.java

```diff
@@ -226,4 +238,27 @@ public boolean awaitClose(long timeout, TimeUnit timeUnit) throws InterruptedExc
         return indicesService.awaitClose(timeout, timeUnit);
     }
 
+    private List<NodeStatsPlugin> getNodeStatsPlugins(final PluginsService pluginsService) {
+        return pluginsService.filterPlugins(NodeStatsPlugin.class).toList();
+    }
+
+    private Optional<PluginNodeStats> getPluginStats() {
+        if (nodeStatsPlugins.isEmpty()) {
+            return Optional.empty();
+        }
+
+        final Map<String, NodeStatsPlugin.Stats> pluginNameToStats = new LinkedHashMap<>();
+        for (final NodeStatsPlugin plugin : nodeStatsPlugins) {
+            final String name = Objects.requireNonNull(plugin.getNodeStatsPluginName());
```

Comment:
minor nit: I am of the opinion that every time we use `Objects.requireNonNull(…)` we should use the helpful error message variant (`Objects.requireNonNull(Object obj, String message)`), because otherwise all we've done is trade `NullPointerException on line X` for `NullPointerException on line Y` when we could have `NullPointerException on line Y: node stats plugins require a non-null name`

(not just here, but everywhere)